### PR TITLE
feat: getElementsByName

### PR DESCRIPTION
**Pull request by tedconn**
Adds support for `Document.getElementsByName`

### CODE REVIEW

File: karma.conf.js [hunk 1]
```diff
@@ -36,7 +36,7 @@ module.exports = function (config) {
       check: {
         global: {
           statements: 100,
-          branches: 91,
+          branches: 100,
```
Comment (tedconn) [inline karma.conf.js:39]: With a misfire on a rebase I reverted the code coverage changes, I will rebase this PR when #44 gets re-merged in

File: src/index.js [hunk 2]
```diff
@@ -281,6 +295,12 @@ function attachSourceIfNecessary ({ nodes }, selector) {
   }
 }
 
+function assertIsDocumentOrShadowRoot (context) {
+  if (context.nodeType !== 11 && context.nodeType !== 9) {
```
Comment (tedconn) [inline src/index.js:299]: I realized duck typing this will be a cleaner approach _and_ work across windows.

File: test/getElementsByName.spec.js [hunk 3]
```diff
@@ -0,0 +1,121 @@
...
+      name = 'email'
+      expected = [
+        {
+          tagName: 'INPUT',
+          classList: []
+        }
+      ]
+      assertResultEqual(name, getElementsByName(name, context), expected, true)
+
+      name = 'test'
```
Comment (nolanlawson) [inline test/getElementsByName.spec.js:86]: Just as a sanity check, can we test for names like `tEsT` and ensure that it doesn't match? (Looks like `getElementsByName` is case-sensitive.)

**Review by nolanlawson — CHANGES_REQUESTED** [review]
LGTM, just a nsmall it about tests

**Review by nolanlawson — APPROVED** [review]

**Pull request merged**